code/client/munkilib/fetch.py

```diff
@@ -92,9 +92,17 @@ def writeCachedChecksum(file_path, fhash=None):
 
 def header_dict_from_list(array):
     """Given a list of strings in http header format, return a dict.
-    If array is None, return None"""
+    A User-Agent header is added if none is present in the list.
+    If array is None, returns a dict with only the User-Agent header."""
+    header_dict = {}
+    machine = munkicommon.getMachineFacts()
+    darwin_version = os.uname()[2]
+    header_dict["User-Agent"] = ("managedsoftwareupdate/%s Darwin/%s (%s) (%s)"
+        % (machine['munki_version'], darwin_version,
+           machine['arch'], machine['machine_model']))
+
     if array is None:
-        return array
+        return header_dict
     header_dict = {}
```

Comment:
So if array is not None, you throw away the User-Agent header? Is that your intent?